### DISCUSSION TITLE
Select with alias

### DISCUSSION THREAD

**Discussion by Suniron** [prisma/prisma#14316] [Q&A] [answered]
Hey there !

Is it possible to have an alias on **select** or **includes**?

**Comment by TasinIshmam** [comment]
Hi @Suniron  

Your question isn't very clear to me. Could you kindly provide an example or more details about what you are looking for?

**Comment by arthurfiorette** [comment, marked as answer]
Hey @Suniron, currently you cannot alias a field returned by the prisma client without [raw database access](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).

You can map the returned object, shaping it according to your needs, before you actually use it:

```ts
function getUser(): Omit<User, 'name'> & { alias: string } {
  return PRISMA.user.findFirst().then(
    (user) => ({ ...user, alias: name, })
  );
}

function getUsers(): Array<Omit<User, 'name'> & { alias: string }> {
  return PRISMA.user.findMany().then(
    (users) => users.map(
      ({ name, ...user }) => ({ ...user, alias: name })
    )
  );
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
